From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hip-ep <hip-ep@amd.com>
Date: Tue, 29 Sep 2026 00:00:00 +0000
Subject: [PATCH 3/3] Demote GCC warning errors in rocMLIR

GCC reports intentional member shadowing in Rock constructors and a
false-positive null dereference in LLVM DenseMap as errors under rocMLIR's
strict warning policy. Keep these diagnostics enabled but non-fatal for GCC.
---
 mlir/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mlir/CMakeLists.txt b/mlir/CMakeLists.txt
index 000000000..000000000 100644
--- a/mlir/CMakeLists.txt
+++ b/mlir/CMakeLists.txt
@@ -49,6 +49,13 @@ add_compile_options(
   $<$<COMPILE_LANGUAGE:CXX>:-Wmissing-declarations>
 )
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  add_compile_options(
+    $<$<COMPILE_LANGUAGE:CXX>:-Wno-error=shadow>
+    $<$<COMPILE_LANGUAGE:CXX>:-Wno-error=null-dereference>
+  )
+endif()
+
 # Refer to https://reviews.llvm.org/D122088
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-- 
2.43.0
